### DISCUSSION TITLE
HD108 support

### DISCUSSION THREAD

**Discussion by r3vEngIT** [awawa-dev/HyperSerialPico#20] [General]
Hey what's up man? My TV broke back in February 2023 so I have been out of the loop with hyperHDR but I see you have some big things going on. I cannot wait to try them out. I went with OLED for the new unit and want to do the lighting even better this time.

I took your recommendation and went with a Plasma 2040 for this build. I was planning on using HD107s this time considering that I did not feel like there was that much of a noticeable improvement when using the HD108s, albeit I never figured out if that was in part to capturing with flatbuffers. Regardless,

1. I am not ready to give up on the HD108s completely and was curious if you planned on adding support for them here in HyperSerialPico?

2. You mentioned Pico boards are really usb 1.1, is this true for the Plasma2040 as well? what kind of limitations does this impose?

3. is it possible to run a second strip with just white LEDs alongside a rgb strip and control them as if it were a rgbw strip? theoretically this would provide the best of both (fast transmission + white color channel).

4. I am surprised you have not developed a board yourself for controlling the LEDs? You have the knowledge and experience to design something that combines performance and features unique to ambient TV lighting. You should do it! haha.

As always, thank you for all your work.

**Comment by awawa-dev** [comment]
Hi
> I am not ready to give up on the HD108s completely and was curious if you planned on adding support for them here in HyperSerialPico?

Implementing this wouldn't be complicated: I just need someone to move the hd108 code from HyperHDR to my LED library for Pico ;) Maybe I'll do it later, I don't have any plans for it yet. Currently, everything in HyperHDR is processed in 24-bit colors anyway, so the only advantage of hd108 over neopixel is lost.

>You mentioned Pico boards are really usb 1.1, is this true for the Plasma2040 as well? what kind of limitations does this impose?

May have conflict with USB2.0 grabber on RPi? Examples of speed calculations and their consequences are available on the blog. In practice, the RP2040 very stably maintains a speed above 4Mb... I do not write here about the declared COM speed, only what it can actually send and receive. esp is worse here, but the possible reason is the Arduino middleware, Pico has almost pure FreeRTOC with TinyUSB.

> is it possible to run a second strip with just white LEDs alongside a rgb strip and control them as if it were a rgbw strip? theoretically this would provide the best of both (fast transmission + white color channel).

Yes you can do that. The question is does it make sense? They will be a bit out of sync and a nightmare to build and configure. If both LED strips are controlled by the AWA controller, thanks to calibration you can, for example, force the sk6812 to display only the white color (common color part) and the second strip to display the rest of the RGB color without the common part (white).

> I am surprised you have not developed a board yourself for controlling the LEDs? You have the knowledge and experience to design something that combines performance and features unique to ambient TV lighting. You should do it! haha.

Pimoroni and Adafruit have already done this. Their RP2040 boards with level shifters are cheap and widely available, it's strange that they haven't undertaken similar projects for ESP32 before, or at least I haven't heard of them. I prefer focus on the software.

**Comment by r3vEngIT** [comment]
I decided to abandon the HD108 and HD107s for RGBW and. . . you were right. the addition of cool white allows near perfect color matching. Scenes with cloudy skies and fields of grass are no longer too blue or too green. Skin tone was one of the hardest colors to match with RGB but with RGBW I am blown away at how well it matches. 

I will say that the vibrance of the colors are not as good as the HD strips, but this could be due to the quality of the strip that I am using. I wanted to use a 96 led/m strip this build and do not regret that decision at all. 96/m is a perfect balance between 60/m and 144/m but I could only find them on Aliexpress from an unknown manufacturer. 

One thing I loved about my HD strips was the fpcb was thick 2oz copper. Do you know of any places that have RGBW in 96/m like this?

**Comment by awawa-dev** [comment]
@r3vEngIT I purchased mine here: https://aliexpress.com/item/32664925043.html  The LED strip has been working for almost 1.5 years without any problems and is used very intensively, although, as always, in such cases I am not responsible for the offer, because on Aliexpress everything can change with the next delivery in stock.

> the addition of cool white allows near perfect color matching. Scenes with cloudy skies and fields of grass are no longer too blue or too green. Skin tone was one of the hardest colors to match with RGB but with RGBW I am blown away at how well it matches.

yes, I also noticed that there are such elements that immediately attract our attention, probably our brains have encoded them through the process of evolution ;) While the Cold RGBW version copes very well with them but for Neutral RGBW I was not entirely satisfied with the way the sky was presented, regardless of calibration, it was always a bit too warm. Same as snow or ice.

**Comment by Anderl94** [comment]
> Implementing this wouldn't be complicated: I just need someone to move the hd108 code from HyperHDR to my LED library for Pico ;) Maybe I'll do it later, I don't have any plans for it yet. Currently, everything in HyperHDR is processed in 24-bit colors anyway, so the only advantage of hd108 over neopixel is lost.

Is there currently support planned for HD108 for Pico?

**Reply by awawa-dev** [reply]
I have a few ideas for HD108, but in the case of Adalight, to use its capabilities and not just to downgrade 48-bit HD108 to regular 24-bit 'APA102', it would require deep changes and it's hard to do something like that if I don't have HD108 myself.

**Reply by r3vEngIT** [reply]
I can send you a strip for free if you want

**Reply by awawa-dev** [reply]
If you want to give a sample of HD108 to me for testing and you are able to send the LED strip within the EU to Poland (otherwise VAT and customs clearance will probably kill the idea), I would be very grateful :-)

**Reply by r3vEngIT** [reply]
would it be easier to put them on eBay for like $1?

**Reply by r3vEngIT** [reply]
i would still like to get these to you but dont know how. here is the link to my eBay listing. contact me on there.
https://www.ebay.com/itm/203672373925

**Reply by awawa-dev** [reply]
I'm sorry, I don't know how to go about it now, there are too many things going on right now :/ And the customs office in my country is messed up and can value things in its own way even if they are declared as $1 or gift + add a customs service fee which depends on the operator perhaps many times the value of inexpensive items.

**Reply by Anderl94** [reply]
I still have a piece of HD108 left and could send it to you from Germany.

**Comment by r3vEngIT** [comment]
i understand your situation. if you could provide a relative location i
would still like to calculate the cost of sending them to you and if not
too high i will cover the cost.

**Comment by r3vEngIT** [comment]
send me a DM with where to send the strip(s) and i will work out the rest.
also, how many do you need for your television? 60/m or 144/m?
